### PR TITLE
Ea/aa ra

### PR DESCRIPTION
**Pull request by practicalint**
Adding Active/Active to 3 major clouds Reference Architectures, with some other clean-up/updating.

### CODE REVIEW

**Review by ml4 — APPROVED** [review]
LGTM

File: content/source/docs/enterprise/before-installing/reference-architecture/aws.md [hunk 1]
```diff
@@ -212,26 +223,34 @@ There is not currently a full monitoring guide for Terraform Enterprise. The fol
...
 Architecture is designed to handle different failure scenarios with
-different probabilities. As the architecture evolves it may provide a
+different probabilities. As the architecture evolves it will provide a
 higher level of service continuity.
 
-#### Single EC2 Instance Failure
+### Terraform Enterprise Server
 
-In the event of the *Terraform Enterprise* instance failing in a way that AWS can
+By utilizing an Auto Scaling Group, a Terraform Enterprise instance automatically recovers
```
Comment (sudomateo) [inline content/source/docs/enterprise/before-installing/reference-architecture/aws.md:237]:
```suggestion
By utilizing an Auto Scaling Group, a Terraform Enterprise instance can automatically recover
```

File: content/source/docs/enterprise/before-installing/reference-architecture/aws.md [hunk 2]
```diff
@@ -364,3 +337,104 @@ Standard. From the AWS website:
...
+- DNS must be redirected to the Load Balancer acting as the entry point for the infrastructure deployed in the secondary AWS Region.
+
+## Active/Active Implementation Mode
+
+### Overview
+
+As stated previously, the _Active/Active_ implementation mode is an extension of the _Standalone_ implementation mode that increases the scalability and load capacity of the Terraform Enterprise platform. The same application runs on multiple Terraform Enterprise instances utilizing the same external services in a shared model. The primary architectural and implementation differences for _Active/Active_ are:
+
+- It can only be run in the _External Services_ mode.
+- The second/alternate and additional nodes are active and processing work at all times.
```
Comment (sudomateo) [inline content/source/docs/enterprise/before-installing/reference-architecture/aws.md:369]:
```suggestion
- The additional nodes are active and processing work at all times.
```

File: content/source/docs/enterprise/before-installing/reference-architecture/aws.md [hunk 3]
```diff
@@ -364,3 +337,104 @@ Standard. From the AWS website:
...
+- There is an addition to the existing external services which is a memory cache which is currently implemented with cloud native implementations of Redis. This is used for the processing queue for the application and has been moved from the individual instance to be a shared resource that manages distribution of work.
```
Comment (sudomateo) [inline content/source/docs/enterprise/before-installing/reference-architecture/aws.md:370]:
```suggestion
- In addition to the existing external services, there is a memory cache which is currently implemented with cloud native implementations of Redis. This is used for the processing queue for the application and has been moved from the individual instance to be a shared resource that manages distribution of work.
```

File: content/source/docs/enterprise/before-installing/reference-architecture/aws.md [hunk 4]
```diff
@@ -364,3 +337,104 @@ Standard. From the AWS website:
...
+
+![aws-aa-infrastructure-diagram](./assets/RA-TFE-AA-AWS-SingleRegion.png)
+
+The above diagram shows the infrastructure components of an _Active/Active_ implementation at a high-level.
+
+### Infrastructure Requirements
+
+#### Active Nodes
+
+The diagram depicts two active nodes to be concise. Additional nodes can be added by simply launching another instance with the identical configuration that points to the same shared external services. The number and sizing of nodes should be based on load requirements and redundancy needs. Nodes should be deployed in alternate zones to accommodate zone failure.
```
Comment (sudomateo) [inline content/source/docs/enterprise/before-installing/reference-architecture/aws.md:391]: Not always! Customers using node IP addresses in their configuration will need to make adjustments.

File: content/source/docs/enterprise/before-installing/reference-architecture/azure.md [hunk 5]
```diff
@@ -202,49 +211,35 @@ are routed to the highly available infrastructure supporting Azure Storage.
...
+scenarios that have different probabilities. As the architecture evolves it will provide a
+higher level of service continuity.
 
-In the event of the active instance failing, the Load Balancer
-should be reconfigured (manually or automatically) to route all traffic
-to the standby instance.
+### Terraform Enterprise Server
 
-~> **Important:** Active-active configuration is not supported due to a serialisation requirement in the core components of Terraform Enterprise; therefore, all traffic from the Load Balancer *MUST* be routed to a single instance.
+By utilizing an VM Scale Set, a Terraform Enterprise instance automatically recovers
```
Comment (sudomateo) [inline content/source/docs/enterprise/before-installing/reference-architecture/azure.md:232]:
```suggestion
By utilizing an VM Scale Set, a Terraform Enterprise instance can automatically recover
```

File: content/source/docs/enterprise/before-installing/reference-architecture/azure.md [hunk 6]
```diff
@@ -299,44 +325,80 @@ services such as DNS.
...
 
-When using the _External Services_ operational mode (PostgreSQL Database and Object Storage), there is still some application configuration data present on the
-Terraform Enterprise server such as installation type, database connection settings, and
-hostname; however, this data rarely changes. We recommend
-[configuring automated
-snapshots](../../admin/automated-recovery.html#configure-snapshots)
-for this installation data so it can be recovered in the event of data
-corruption.
+- It can only be run in the _External Services_ mode.
+- The second/alternate and additional nodes are active and processing work at all times.
```
Comment (sudomateo) [inline content/source/docs/enterprise/before-installing/reference-architecture/azure.md:335]:
```suggestion
- The additional nodes are active and processing work at all times.
```

File: content/source/docs/enterprise/before-installing/reference-architecture/azure.md [hunk 7]
```diff
@@ -299,44 +325,80 @@ services such as DNS.
...
+- There is an addition to the existing external services which is a memory cache which is currently implemented with cloud native implementations of Redis. This is used for the processing queue for the application and has been moved from the individual instance to be a shared resource that manages distribution of work.
```
Comment (sudomateo) [inline content/source/docs/enterprise/before-installing/reference-architecture/azure.md:336]:
```suggestion
- In an addition to the existing external services, there is a memory cache which is currently implemented with cloud native implementations of Redis. This is used for the processing queue for the application and has been moved from the individual instance to be a shared resource that manages distribution of work.
```

File: content/source/docs/enterprise/before-installing/reference-architecture/azure.md [hunk 8]
```diff
@@ -299,44 +325,80 @@ services such as DNS.
...
+
+![aws-aa-infrastructure-diagram](./assets/RA-TFE-AA-Azure-SingleRegion.png)
+
+The above diagram shows the infrastructure components of an _Active/Active_ implementation at a high-level.
+
+### Infrastructure Requirements
+
+#### Active Nodes
+
+The diagram depicts two active nodes to be concise. Additional nodes can be added by simply launching another instance with the identical configuration that points to the same shared external services. The number and sizing of nodes should be based on load requirements and redundancy needs. Nodes should be deployed in alternate zones to accommodate zone failure.
```
Comment (sudomateo) [inline content/source/docs/enterprise/before-installing/reference-architecture/azure.md:357]: Not always! Customers using node IP addresses in their configuration will need to make adjustments.

**Review by sudomateo — COMMENTED** [review]
Same feedback applies to GCP as well.

**Review by sudomateo — APPROVED** [review]
Will give my approval as the comments I made are non-blocking.

File: content/source/docs/enterprise/before-installing/reference-architecture/aws.md (hunk 4 above)
Comment (practicalint) [reply content/source/docs/enterprise/before-installing/reference-architecture/aws.md:391]: altered language but not in detail

File: content/source/docs/enterprise/before-installing/reference-architecture/azure.md (hunk 8 above)
Comment (practicalint) [reply content/source/docs/enterprise/before-installing/reference-architecture/azure.md:357]: altered language but not in detail
All changes reflected in GCP also. Committed under new branch and new PR

**Pull request merged**